services/blockchain-indexer/shared/dataService/business/events.js

```diff
@@ -211,20 +210,8 @@ const getEvents = async (params) => {
 		{ concurrency: eventsInfo.length },
 	);
 
-	let total;
-	const { order, sort, ...remQueryParams } = queryParams;
-
-	if (isTopicInQuery) {
-		total = await eventTopicsTable.count(
-			{ ...remQueryParams, distinct: 'eventID' },
-			['eventID'],
-		);
-	} else {
-		// If params doesn't contain event_topics specific column data
-		// then count all rows of event table for query optimization.
-		const { leftOuterJoin, ...remParamsWithoutJoin } = remQueryParams;
-		total = await eventsTable.count(remParamsWithoutJoin, ['id']);
-	}
+	const { order, sort, ...queryParamsWithoutOrderAndSort } = queryParams;
+	const total = await eventsTable.count({ ...queryParamsWithoutOrderAndSort, distinct: 'id' });
```

Comment:
Do we need the `distinct` here?

```suggestion
	const total = await eventsTable.count({ ...queryParamsWithoutOrderAndSort });
```